experiment/simulation/js/main.js

```diff
@@ -1 +1,268 @@
 //Your JavaScript goes in here
+
+document.addEventListener('DOMContentLoaded', function(){
+
+	var play = true
+
+	var playButton = document.getElementById('play');
+	var pauseButton = document.getElementById('pause');
+	var restartButton = document.getElementById('restart');
+	var submitButton = document.getElementById('submit');
+
+	pauseButton.addEventListener('click', function() { window.clearTimeout(tmHandle) });
+	playButton.addEventListener('click', function() {  window.clearTimeout(tmHandle); tmHandle = setTimeout(draw, 1000 / fps); });
+	restartButton.addEventListener('click', function() {restart()})
+
+	function restart() 
+	{ 
+		window.clearTimeout(tmHandle) 
+
+		bldg = [
+			[[upL[0], defY], [upR[0], defY], [startR[0], defY + height], [startL[0], defY + height]],
+			[[upL[1], defY], [upR[1], defY], [startR[1], defY + height], [startL[1], defY + height]],
+			[[upL[2], defY], [upR[2], defY], [startR[2], defY + height], [startL[2], defY + height]]
+		]
+
+		ground = [
+			[startL[0] - 50, defY + height + 40],
+			[startL[0], defY + height - 40],
+			[startR[2] + 50, defY + height - 40],
+			[startR[2], defY + height + 40],
+		]
+
+		layer2 = [
+			{...ground[0]},
+			[ground[0][0], defY + height + 40 + thickness],
+			[startR[2] + thickness, defY + height + 40 + thickness],
+			[ground[2][0] + thickness, defY + height - 40 + thickness],
+			{...ground[2]},
+			{...ground[3]},
+		]
+
+		dirn = -1;
+		tmHandle = window.setTimeout(draw, 1000 / fps); 
+	}
+
+	var slider_hei = document.getElementById("height");
+	var output_hei = document.getElementById("demo_height");
+	output_hei.innerHTML = slider_hei.value; // Display the default slider value
+
+	// Update the current slider value (each time you drag the slider handle)
+	slider_hei.oninput = function() {
+		output_hei.innerHTML = this.value;
+	}
+
+	var slider_mot = document.getElementById("motion");
+	var output_mot = document.getElementById("demo_motion");
+	output_mot.innerHTML = slider_mot.value; // Display the default slider value
+
+	// Update the current slider value (each time you drag the slider handle)
+	slider_mot.oninput = function() {
+		output_mot.innerHTML = this.value;
+	}
+
+	submitButton.addEventListener('click', function() {
+		height = Number(document.getElementById("height").value)
+		vibe = Number(document.getElementById("motion").value)
+		restart()
+	});
+
+	function curvedArea(ctx, e, gradX, gradY)
+	{
+		ctx.bezierCurveTo(e[0], e[1] += gradY, e[0] += gradX, e[1] += gradY, e[0] += gradX, e[1]);
+		ctx.bezierCurveTo(e[0] += gradX, e[1], e[0] += gradX, e[1] -= gradY, e[0], e[1] -= gradY);
+	}
+
+	function updateGround(ground, layer2, chg)
+	{
+		for(let i = 0; i < ground.length; ++i)
```

Comment:
Prefer to use more semantic function like forEach, map, reduce etc. In case you must use a for loop, please use {..} to encapsulate the code.